doc/add_dash_anchors.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Add Dash-style anchors to already-generated HTML documentation.
+
+This script iterates over pre-specified HTML files generated via
+sphinx-build, finds all of the sections, and adds Dash-style anchors
+so that when those HTML files are displayed in the Dash macOS app,
+the sections are displayed in a Dash TOC on the right.
+
+:author: Nitin Madnani (nmadnani@ets.org)
+:organization: ETS
+"""
+
+import argparse
+import logging
+import re
+import pathlib
+import unicodedata
+from urllib.parse import quote
+
+from bs4 import BeautifulSoup
+
+# pre-define the list of HTML files we want to miodify
+FILES_TO_MODIFY = ["advanced_usage.html", "api.html", "contributing.html",
```

Comment:
How do we make sure this list is up-to-date? Can we create it automatically based on directory content?

Reply:
Yeah, I tried this but struggled with it. There are some auto-generated HTML files that we do not want to add Dash nchors to (for example, `genindex.html`). I haven’t been able to figure out if this set is fixed and so we can just include all files _except_ the ones in this set. I can work on this in the future.